deps/rabbitmq_management_agent/src/rabbit_mgmt_metrics_gc.erl

```diff
@@ -137,18 +140,23 @@ delete_samples(Table, Id, Intervals) ->
     [ets:delete(Table, {Id, I}) || I <- Intervals],
     ok.
 
-index_delete(consumer_stats = Table, channel = Type, Id) ->
-    IndexTable = rabbit_mgmt_metrics_collector:index_table(Table, Type),
-    MatchPattern = {'_', Id, '_'},
-    %% Delete consumer_stats_queue_index
-    ets:match_delete(consumer_stats_queue_index,
-                     {'_', MatchPattern}),
-    %% Delete consumer_stats
-    ets:match_delete(consumer_stats,
-                     {MatchPattern,'_'}),
-    %% Delete consumer_stats_channel_index
-    ets:delete(IndexTable, Id),
-    ok;
+index_delete(consumer_stats = Table, {channel = Type, Count}, Id) ->
+    case Count > ?LARGE_CONSUMER_COUNT of
```

Comment:
We can add a comment explaining the condition here: `ets:match_delete/2` is acceptable or optimal for a large number of entries **on this particular channel** but when there are many channels with a few (very often just one) consumers each, its sequential scans become prohibitively expensive.